templates/layouts/_questionnaire.html

```diff
@@ -21,6 +21,8 @@
       </div>
   {% endblock %}
 
+  {% block cancel_and_return %}{% endblock %}
```

Comment:
This is defined in `listaddquestion.html`, `listeditquestion.html` and `listremovequestion.html` but should only need to define this once and then can be used in `questionnaire.html`.

Reply:
I have abstracted the code for these blocks.